pkg/dogstatsd/server.go

```diff
@@ -289,6 +289,14 @@ func NewServer(demultiplexer aggregator.Demultiplexer, extraTags []string) (*Ser
 		extraTags = config.Datadog.GetStringSlice("dogstatsd_tags")
 	}
 
+	// if the server is running in a context where static tags are required, add those
+	// to extraTags.
+	if staticTags := util.GetStaticTagsSlice(context.TODO()); staticTags != nil {
+		extraTags = append([]string{}, extraTags...)
```

Comment:
I might be missing something, but why do you copy the `extraTags` slice? 🤔 

If it's because you're unsure whether you can mutate what `config.Datadog.GetStringSlice` returns, then you'd want to copy the slice outside of this `if` block since you then sort its elements. But in any case `config.Datadog.GetStringSlice` should return a new slice so you can mutate it.

Reply:
Filed https://github.com/DataDog/datadog-agent/pull/10968